modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/ClassifyByRangeProcess.java

```diff
@@ -0,0 +1,144 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2011-2017, Open Source Geospatial Foundation (OSGeo)
+ *    (C) 2008-2011 TOPP - www.openplans.org.
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.process.vector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.filter.function.CategorizeFunction;
+import org.geotools.filter.function.RangedClassifier;
+import org.geotools.process.ProcessException;
+import org.geotools.process.factory.DescribeParameter;
+import org.geotools.process.factory.DescribeProcess;
+import org.geotools.process.factory.DescribeResult;
+import org.geotools.process.vector.TransformProcess.Definition;
+import org.geotools.util.Converters;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.filter.FilterFactory;
+import org.opengis.filter.expression.Expression;
+import org.opengis.filter.expression.Function;
+
+/**
+ * Computes a new attribute to classify another attribute by intervals over vector data sets.
+ * 
+ * @author Mauro Bartolomeoli
+ *
+ * @source $URL$
+ */
+@DescribeProcess(title = "ClassifyByRange", description = "Computes a new attribute to classify another attribute by intervals over vector data sets.")
+public class ClassifyByRangeProcess implements VectorProcess {
+    
+    static FilterFactory filterFactory = CommonFactoryFinder.getFilterFactory(null);
+    
+    @DescribeResult(name = "result", description = "Classified feature collection")
+    public SimpleFeatureCollection execute(
+            @DescribeParameter(name = "features", description = "Input feature collection") SimpleFeatureCollection features,
+            @DescribeParameter(name = "classifyOnAttribute", description = "Attribute to be classified using intervals of values.") String classifyOnAttribute,
+            @DescribeParameter(name = "thresholds", min = 0, description = "Comma delimited list of thresholds (use this one to specify custom intervals).") String thresholds,
```

Comment:
A list of things should be either String[] or List<String> (the latter requires a one more annotation attribute to specify what's inside the list

Reply:
Also, best to specify what happens if both custom thresholds and classes count are specified at the same time